components/plan-build/bin/hab-plan-build.sh

```diff
@@ -2318,28 +2318,63 @@ SRC_PATH="$PLAN_CONTEXT"
 # Expand the path of this program to an absolute path
 THIS_PROGRAM=$(abspath "$0")
 
-# Now to ensure a `plan.sh` exists where we expect it. There are 2 possible
-# candidate locations relative to the `$PLAN_CONTEXT` directory: a `./plan.sh`
-# or a `./habitat/plan.sh`. Only one or the other location is allowed so that
-# a Plan author isn't confused if they edit one to have this program read
-# the other.
+# Now to ensure a `plan.sh` exists where we expect it. There are 4 possible
+# candidate locations relative to the `$PLAN_CONTEXT` directory: 
+#   `./plan.sh`
+#   `./habitat/plan.sh`
+#   `./$pkg_target/plan.sh
+#   `./habitat/$pkg_target/plan.sh 
+# In most cases, only one location is allowed so that a Plan author isn't 
```

Comment:
I'd change this to read something like: `In most cases, a plan author should use the default plan location, be it ./plan.sh or ./$pkg_target/plan.sh. In the even they need to create or modify a variant, they should place the variant plan file in the appropriate /$pkg_target/ directory relative to their default plan location.`